Index: target/linux/atheros/files/include/asm-mips/mach-atheros/ar5315/ar5315.h
===================================================================
--- target/linux/atheros/files/include/asm-mips/mach-atheros/ar5315/ar5315.h	(revision 16491)
+++ target/linux/atheros/files/include/asm-mips/mach-atheros/ar5315/ar5315.h	(working copy)
@@ -355,7 +355,7 @@
 #define AR5315_GPIO_INT_LVL_HIGH			2   /* High Level Triggered */
 #define AR5315_GPIO_INT_LVL_EDGE			3   /* Edge Triggered */
 
-#define AR5315_RESET_GPIO       5
+#define AR5315_RESET_GPIO       6
 #define AR5315_NUM_GPIO         22
 
 
Index: target/linux/atheros/config-2.6.26
===================================================================
--- target/linux/atheros/config-2.6.26	(revision 16491)
+++ target/linux/atheros/config-2.6.26	(working copy)
@@ -20,7 +20,7 @@
 # CONFIG_BT is not set
 CONFIG_CEVT_R4K=y
 CONFIG_CLASSIC_RCU=y
-CONFIG_CMDLINE="console=ttyS0,9600 rootfstype=squashfs,jffs2 init=/etc/preinit"
+CONFIG_CMDLINE="console=ttyS0,4800 rootfstype=squashfs,jffs2 init=/etc/preinit"
 CONFIG_CPU_BIG_ENDIAN=y
 CONFIG_CPU_HAS_LLSC=y
 CONFIG_CPU_HAS_PREFETCH=y
