? menu.allow.display.of.single.tabs.patch
Index: menu.inc
===================================================================
RCS file: /cvs/drupal/drupal/includes/menu.inc,v
retrieving revision 1.255.2.31
diff -u -p -r1.255.2.31 menu.inc
--- menu.inc	27 Apr 2009 12:50:13 -0000	1.255.2.31
+++ menu.inc	3 Aug 2009 20:19:06 -0000
@@ -1422,8 +1422,8 @@ function menu_local_tasks($level = 0, $r
     return $root_path;
   }
   else {
-    // We do not display single tabs.
-    return (isset($tabs[$level]) && $tabs[$level]['count'] > 1) ? $tabs[$level]['output'] : '';
+    // We do not display single tabs: BVDM yes we do for flukso!
+    return (isset($tabs[$level])) ? $tabs[$level]['output'] : '';
   }
 }
 
