Index: target/linux/generic-2.6/config-2.6.30
===================================================================
--- target/linux/generic-2.6/config-2.6.30	(revision 23099)
+++ target/linux/generic-2.6/config-2.6.30	(working copy)
@@ -1964,7 +1964,7 @@
 # CONFIG_SENSORS_W83793 is not set
 # CONFIG_SENSORS_W83L785TS is not set
 # CONFIG_SENSORS_W83L786NG is not set
-CONFIG_SERIAL_8250_CONSOLE=y
+# CONFIG_SERIAL_8250_CONSOLE is not set
 # CONFIG_SERIAL_8250_CS is not set
 CONFIG_SERIAL_8250_NR_UARTS=2
 # CONFIG_SERIAL_8250_PCI is not set
