Index: target/linux/generic-2.6/config-2.6.26
===================================================================
--- target/linux/generic-2.6/config-2.6.26	(revision 16491)
+++ target/linux/generic-2.6/config-2.6.26	(working copy)
@@ -1359,7 +1359,7 @@
 # CONFIG_SENSORS_W83L785TS is not set
 # CONFIG_SENSORS_W83L786NG is not set
 CONFIG_SERIAL_8250=y
-CONFIG_SERIAL_8250_CONSOLE=y
+# CONFIG_SERIAL_8250_CONSOLE is not set
 CONFIG_SERIAL_8250_CS=m
 CONFIG_SERIAL_8250_NR_UARTS=2
 # CONFIG_SERIAL_8250_PCI is not set
