diff -urNad trunk~/src/lxplib.c trunk/src/lxplib.c
--- trunk~/src/lxplib.c	2006-03-20 20:55:53.000000000 +0100
+++ trunk/src/lxplib.c	2006-09-03 10:10:14.000000000 +0200
@@ -13,7 +13,6 @@
 
 #include "lua.h"
 #include "lauxlib.h"
-#include "compat-5.1.h"
 
 
 #include "lxplib.h"
@@ -530,8 +529,8 @@
   lua_pushliteral(L, "__index");
   lua_pushvalue(L, -2);
   lua_rawset(L, -3);
-  luaL_openlib (L, NULL, lxp_meths, 0);
-  luaL_openlib (L, "lxp", lxp_funcs, 0);
+  luaL_register (L, NULL, lxp_meths);
+  luaL_register (L, "lxp", lxp_funcs);
   set_info (L);
 
   return 1;
