--- a/config.mk	2013-04-23 20:22:12.356342078 +0200
+++ b/config.mk	2013-04-23 20:22:50.324341248 +0200
@@ -23,7 +23,7 @@
 # Comment out to disable TLS/PSK support in the broker and client. Requires
 # WITH_TLS=yes.
 # This must be disabled if using openssl < 1.0.
-WITH_TLS_PSK:=yes
+WITH_TLS_PSK:=no
 
 # Comment out to disable client client threading support.
 WITH_THREADING:=yes
@@ -105,7 +105,7 @@
 LIB_LIBS:=
 PASSWD_LIBS:=
 
-CLIENT_LDFLAGS:=$(LDFLAGS) -L../lib ../lib/libmosquitto.so.${SOVERSION}
+CLIENT_LDFLAGS:=$(LDFLAGS) -L../lib ../lib/libmosquitto.so.${SOVERSION} -lcrypto -lssl
 
 ifeq ($(UNAME),SunOS)
 	ifeq ($(CC),cc)
