--- avrdude-5.10/config_gram.y.orig	2010-11-05 11:01:14.895247246 +0100
+++ avrdude-5.10/config_gram.y	2010-11-05 11:02:15.219259178 +0100
@@ -1212,10 +1212,10 @@
 
   value = v->value.number;
 
-  if ((value <= 0) || (value >= 18)) {
+  if ((value < 0) || (value >= 18)) {
     fprintf(stderr, 
             "%s: error at line %d of %s: pin must be in the "
-            "range 1-17\n",
+            "range 0-17\n",
             progname, lineno, infile);
     exit(1);
   }
